Assets/Scripts/Controllers/CharacterController.cs

```diff
@@ -25,7 +25,20 @@ private void FixedUpdate()
 
     public void OnMove(InputValue value)
     {
-        _movement = value.Get<Vector2>();
+        // check if vector2.up is present, we don't use that yet so we odn't need it at all
+        if (value.Get<Vector2>().y > 0)
+        {
+            //if up is present, reset y to 0
+            Vector2 move = value.Get<Vector2>();
+            move.y = 0;
+            _movement = move;
+        }
+        else
+        {
+            _movement = value.Get<Vector2>();
+        }
```

Comment:
Instead of doing this check, we can just _always_ set the y component to zero:

```cs
public void OnMove(InputValue value)
{
  _movementInput = value.Get<Vector2>();
  _movementInput.y = 0f;
}
```
This is a little more efficient, but is also just more readable.

Reply:
I have Changed the approach by just removing W and Up Arrow from the input asset, Now we don't even have to check for it. and W and up arrow are free to consider for other future functions